AirCastingTests/Logic/SDSync/SdSyncAveragingServiceTests.swift

```diff
@@ -0,0 +1,135 @@
+// Created by Lunar on 13/10/2022.
+//
+
+import XCTest
+@testable import AirCasting
+
+extension SDSyncMeasurement: Equatable {
+    public static func == (lhs: AirCasting.SDSyncMeasurement, rhs: AirCasting.SDSyncMeasurement) -> Bool {
+        lhs.measuredAt == rhs.measuredAt && lhs.value == rhs.value && lhs.location?.latitude == rhs.location?.latitude && lhs.location?.longitude == rhs.location?.longitude
+    }
+}
+
+final class SdSyncAveragingServiceTests: XCTestCase {
+    let service = DefaultSDSyncAveragingService()
+
+    func test_averageMeasurementsWithReminderCalled_returnsUnaveragedMeasurementsAsReminder() throws {
+        let date = Date()
+        
+        let measurements = [SDSyncMeasurement(measuredAt: date, value: 3)]
+        
+        let reminder = service.averageMeasurementsWithReminder(measurements: measurements, startTime: date, averagingWindow: .firstThresholdWindow, action: { _, _ in })
+        XCTAssertEqual(reminder, measurements)
+    }
+    
+    func test_averageMeasurementsWithReminderCalled_callsCompletionWithAnAveragedMeasurement() throws {
+        let thresholdWindow = AveragingWindow.firstThresholdWindow
+        let startDate = Date.init(timeInterval: TimeInterval(-thresholdWindow.rawValue * 5), since: Date())
+        
+        let measurements = [SDSyncMeasurement(measuredAt: startDate, value: 2),
+                            SDSyncMeasurement(measuredAt: startDate+1, value: 1),
+                            SDSyncMeasurement(measuredAt: startDate+TimeInterval(thresholdWindow.rawValue), value: 3)]
+        
+        var averagedMeasurement: SDSyncMeasurement?
+        
+        let reminder = service.averageMeasurementsWithReminder(measurements: measurements, startTime: startDate, averagingWindow: .firstThresholdWindow, action: { measurement, _ in averagedMeasurement = measurement })
+        
+        XCTAssertEqual(reminder, [SDSyncMeasurement(measuredAt: startDate+TimeInterval(thresholdWindow.rawValue), value: 3)])
+        XCTAssertEqual(averagedMeasurement, SDSyncMeasurement(measuredAt: startDate + TimeInterval(thresholdWindow.rawValue) - 1, value: 1.5))
+    }
+    
+    func test_averageMeasurementsWithReminderCalled_callsCompletionWithMeasurementsFromInterval() throws {
```

Comment:
again completion here, but I don't have any ideas honestly. I will try to think about something tomorrow.